src/olympia/versions/tasks.py

```diff
@@ -52,22 +111,30 @@ def generate_static_theme_preview(theme_manifest, header_root, preview):
     context.update(theme_manifest.get('colors', {}))
     header_url = theme_manifest.get('images', {}).get('headerURL')
 
-    header_path = os.path.join(header_root, header_url)
-    try:
-        with storage.open(header_path, 'rb') as header_file:
-            header_blob = header_file.read()
-            with Image.open(StringIO.StringIO(header_blob)) as header_image:
-                (width, height) = header_image.size
-                context.update(header_src_height=height)
-                meetOrSlice = ('meet' if width < amo.THEME_PREVIEW_SIZE.width
-                               else 'slice')
-                context.update(
-                    preserve_aspect_ratio='xMaxYMin %s' % meetOrSlice)
-            data_url = 'data:image/%s;base64,%s' % (
-                header_image.format.lower(), b64encode(header_blob))
-            context.update(header_src=data_url)
-    except IOError as io_error:
-        log.debug(io_error)
+    header_src, header_width, header_height = encode_header_image(
+        os.path.join(header_root, header_url))
+    meetOrSlice = ('meet' if header_width < amo.THEME_PREVIEW_SIZE.width
```

Comment:
didn't catch it before, but this should be `meet_or_slice` to stay consistent

Reply:
heh, conversion from javascript.